lib/ansible/modules/storage/hpe/hpe3par_cpg.py

```diff
@@ -0,0 +1,435 @@
+#
+# You should have received a copy of the GNU General Public License along
+# with this program.  If not, see <https://www.gnu.org/licenses/>
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+author: "Hewlett Packard Enterprise (ecostor@groups.ext.hpe.com)"
+description:
+  - "Create and delete CPG on HPE 3PAR."
+module: hpe3par_cpg
+options:
+  cpg_name:
+    description:
+      - "Name of the CPG."
+    required: true
+  disk_type:
+    choices:
+      - FC
+      - NL
+      - SSD
+    description:
+      - "Specifies that physical disks must have the specified device type."
+    required: false
+  domain:
+    description:
+      - "Specifies the name of the domain in which the object will reside."
+    required: false
+  growth_increment:
+    default: -1.0
+    description:
+      - "Specifies the growth increment the amount of logical disk storage
+       created on each auto-grow operation.\n"
+    required: false
+  growth_increment_unit:
+    choices:
+      - MiB
+      - GiB
+      - TiB
+    default: GiB
+    description:
+      - "Unit of growth increment."
+    required: false
+  growth_limit:
+    default: -1.0
+    description:
+      - "Specifies that the autogrow operation is limited to the specified
+       storage amount that sets the growth limit.\n"
+    required: false
+  growth_limit_unit:
+    choices:
+      - MiB
+      - GiB
+      - TiB
+    default: GiB
+    description:
+      - "Unit of growth limit."
+    required: false
+  growth_warning:
+    default: -1.0
+    description:
+      - "Specifies that the threshold of used logical disk space when exceeded
+       results in a warning alert.\n"
+    required: false
+  growth_warning_unit:
+    choices:
+      - MiB
+      - GiB
+      - TiB
+    default: GiB
+    description:
+      - "Unit of growth warning."
+    required: false
+  high_availability:
+    choices:
+      - PORT
+      - CAGE
+      - MAG
+    description:
+      - "Specifies that the layout must support the failure of one port pair,
+       one cage, or one magazine.\n"
+    required: false
+  raid_type:
+    choices:
+      - R0
+      - R1
+      - R5
+      - R6
+    description:
+      - "Specifies the RAID type for the logical disk."
+    required: false
+  set_size:
+    default: -1
+    description:
+      - "Specifies the set size in the number of chunklets."
+    required: false
+  state:
+    choices:
+      - present
+      - absent
+    description:
+      - "Whether the specified CPG should exist or not."
+    required: true
+extends_documentation_fragment: hpe3par
+short_description: "Manage HPE 3PAR CPG"
+version_added: "2.6"
+'''
+
+
+EXAMPLES = r'''
+    - name: Create CPG "{{ cpg_name }}"
+      hpe3par_cpg:
+        storage_system_ip="{{ storage_system_ip }}"
```

Comment:
It is a lot easier for users if examples show real-life content, rather than stub templates.
Now the example does not really add anything.

Also, it should be using the YAML syntax, and not the old key=value syntax.

Reply:
Default connection parameters are called "host", "username", "password", I think it helps if your module is consistent with the standard parameter names. If you still like, you can add aliases, but I don't see the value.

Reply:
Modified the examples to show real life contents as mentioned.

In case of 3PAR, host has a different meaning where the volumes are presented. Therefore, keeping it as is to keep it unambiguous